### ISSUE TITLE
Sass 1.59 doesn't compile on Node.js 12

### ISSUE THREAD

**Issue by jlenos** [sass/dart-sass#1914]
Node.js version: 12.22.12
Function used: `renderSync`

```
/home/project/node_modules/sass/sass.dart.js:19
self.exports = _cliPkgExportParam ?? exports;
                                   ^

SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Object.<anonymous> (/home/project/node_modules/sass/sass.default.js:1:15)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
exec: /home/project/node_modules/sass/sass.dart.js:19
self.exports = _cliPkgExportParam ?? exports;
                                   ^
```

**Comment by ntkme** [comment]
From link above:

> Once a Node.js version is out of LTS, Dart Sass considers itself free to break support if necessary.

**Comment by ntkme** [comment]
To be fair, if you're trying to keep your dependency up to date, you should really update node version as 12 LTS was end of life 30 Apr 2022.

**Comment by jlenos** [comment]
> From link above:
> 
> > Once a Node.js version is out of LTS, Dart Sass considers itself free to break support if necessary.

This is against semantic versioning principles, dictating that breaking changes should be major version bumps.

I had a carret on my Sass dependency which meant that I was trusting it not to do a breaking change. With this mentality that Sass has, where you can drop Node.js support at a whim, you imply that we should always pin a specific version of Sass. This is the only npm dependency I know of that doesn't do a major change when dropping support for a Node.js version support

**Comment by ntkme** [comment]
> This is against semantic versioning principles, dictating that breaking changes should be major version bumps.

Agreed, but it is also explicitly documented.

The Sass project is in a dilemma that it has scheduled tons of deprecations that are supposed to dropped in 2.x, with all the communications and documentations already in place. However, it is still not really to roll them all out today. Thus, cutting 2.x just for node 12 EOL and not rolling out other changes will cause even more confusion. On the other hand, keep supporting an EOL node version because we cannot cut 2.x yet is a huge burden on all Sass maintainers and contributors.

**Comment by jlenos** [comment]
Ok, I will just pin my dependency to 1.58.3 for now until I upgrade my projects to higher Node.js versions.

Thanks for your replies and I hope the progress goes well on 2.x 🙂

**Comment by jathak** [comment]
We inadvertently started using JS features that were unsupported in Node 12 without declaring we were dropping support  in our package.json. We've now published a new version (1.59.3) that should be compatible with Node 12 again.

While we do plan to drop support for Node 12 soon, since it's EOL, we shouldn't have done so without updating the package.json, and we're sorry for the trouble.
